codec/src/main/java/io/netty/handler/codec/DefaultHeaders.java

```diff
@@ -1073,14 +1083,18 @@ public V next() {
             if (!hasNext()) {
                 throw new NoSuchElementException();
             }
-            HeaderEntry<K, V> current = next;
+            previous = next;
             calculateNext(next.next);
-            return current.value;
+            return previous.value;
         }
 
         @Override
         public void remove() {
-            throw new UnsupportedOperationException("read only");
+            if (previous == null) {
+                throw new IllegalStateException();
```

Comment:
I am not sure how to feel about this honestly as it is not really the API contract of an `Iterator`.

Reply:
IIUC this is required by the API contract: 

```java
     * @throws IllegalStateException if the {@code next} method has not
     *         yet been called, or the {@code remove} method has already
     *         been called after the last call to the {@code next}
     *         method
```

Reply:
TIL.... never noticed this before in the `Iterator` javadocs.

Reply:
As this is required can you also add a test case for it ?

Reply:
done